pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -550,18 +550,55 @@ func (service *SubnetPortService) AllocatePortFromSubnet(subnet *model.VpcSubnet
 
 	info.lock.Lock()
 	defer info.lock.Unlock()
+
+	dhcpMode := "DHCP_DEACTIVATED"
+	subnetInfo, _ := servicecommon.ParseVPCResourcePath(*subnet.Path)
+	if subnet.SubnetDhcpConfig != nil && subnet.SubnetDhcpConfig.Mode != nil {
+		dhcpMode = *subnet.SubnetDhcpConfig.Mode
+	}
+	// For DHCP Server mode Subnet, get total IPs from DHCP IP Pool from NSX each time
+	// since user might update reservedIPRanges for the subnet and it impacts the DHCP Pool size
+	if dhcpMode == "DHCP_SERVER" {
+		dhcpServerStats, err := service.NSXClient.StatsClient.Get(subnetInfo.OrgID, subnetInfo.ProjectID, subnetInfo.VPCID, subnetInfo.ID, nil, nil, nil, nil, nil, nil, nil)
+		if err != nil {
+			log.Error(err, "Failed to get Subnet dhcp-server-config stats", "Subnet", *subnet.Path)
+			return false
```

Comment:
Could we return an error in this condition as SubnetSet also use this function to determine whether we need to scale up? We may need to distinguish the API error with no ip in Subnet in the return value.
https://github.com/vmware-tanzu/nsx-operator/blob/2adfb3d89671528f22952e2ada2f74ef468dbd14/pkg/controllers/common/utils.go#L37